tools/python/boutiques/localExec.py

```diff
@@ -474,15 +483,23 @@ def _pullSingImage(self, conName, conIndex, conImage):
         sing_command = "singularity pull --name " + pull_loc
         (stdout, stderr), return_code = self._localExecute(
                                                 sing_command)
+        self._cleanUpAfterSingPull(lockDir)
```

Comment:
Style nitpick: it would be clearer if all the calls to `cleanUpAfterSingPull` would be in the while loop above.